vmm/src/vm.rs

```diff
@@ -453,9 +455,24 @@ impl Vm {
                     .checked_add(KERNEL_64BIT_ENTRY_OFFSET)
                     .ok_or(Error::MemOverflow)?;
 
-                Ok(GuestAddress(load_addr))
+                Ok(EntryPoint {
+                    entry_addr: GuestAddress(load_addr),
+                    protocol: BootProtocol::LinuxBoot,
+                })
             }
             None => {
+                // Assume by default Linux boot protocol is used and not PVH
+                let mut entry_point_addr: GuestAddress = entry_addr.kernel_load;
+
+                let boot_prot = if cfg!(feature = "pvh_boot") && entry_addr.pvh_entry_addr.is_some()
+                {
+                    // entry_addr.pvh_entry_addr field is safe to unwrap here
+                    entry_point_addr = entry_addr.pvh_entry_addr.unwrap();
```

Comment:
Instead of checking if `pvh_entry_addr` is `Some()` and then unwrap it, it's better to do something like this:
```rust
if let Some(pvh_entry_addr) = &entry_addr.pvh_entry_addr {
    entry_point_addr = *pvh_entry_addr;
```

Reply:
I understand that the `if let` construct is more idiomatic, but doesn't using it in this instance while `pvh_boot` is still a non-default feature actually makes this code block longer? 

Since we still need the statement to evaluate to `BootProtocol::LinuxBoot` if `pvh_entry_addr` is `None`, the whole block would look like this:
```rust
let boot_prot = if cfg!(feature = "pvh_boot") {
      if let Some(pvh_entry_addr) = &entry_addr.pvh_entry_addr {
            entry_point_addr = *pvh_entry_addr;
            BootProtocol::PvhBoot
      } else {
            BootProtocol::LinuxBoot
      } 
} else {
      BootProtocol::LinuxBoot 
}
```
Assuming of course there is not a simpler alternative that I am missing entirely. If we decide that PVH should be enabled by default, then that makes things easy. But otherwise the original code is more concise and avoids repetition, even if is a bit awkward in style, no?